src/main/java/org/spongepowered/common/mixin/core/entity/merchant/villager/AbstractVillagerEntityMixin.java

```diff
@@ -0,0 +1,92 @@
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+import org.spongepowered.common.mixin.core.entity.AgeableEntityMixin;
+import org.spongepowered.common.registry.SpongeVillagerRegistry;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Mixin(AbstractVillagerEntity.class)
+public abstract class AbstractVillagerEntityMixin extends AgeableEntityMixin {
+    private List<MerchantOffer> impl$newOffers = new ArrayList<>();
+
+    /**
+     * @author i509VCB - February 17th, 2020 - 1.14.4
+     *
+     * @reason Here we apply the Trade Mutators. For a villager we will apply our mutators and then add them to the villager's merchant offers.
+     * If a wandering trader is the subject here, we do nothing since that merchant would already have it's new merchant offers.
+     *
+     * @todo: Do we want to support Trade Mutators on Wandering Trader's TradeOffers?
+     *
+     * @param givenMerchantOffers The merchant's current merchant offers.
+     * @param newTrades The trade factories representing the new trades this villager will receive.
+     * @param maxNumbers The maximum amount of trades for this villager to receive.
+     * @param ci CallbackInfo.
+     */
+    @Inject(method = "addTrades(Lnet/minecraft/item/MerchantOffers;[Lnet/minecraft/entity/merchant/villager/VillagerTrades$ITrade;I)V", at = @At("TAIL"))
+    private void impl$addAndApplyTradeMutators(MerchantOffers givenMerchantOffers, VillagerTrades.ITrade[] newTrades, int maxNumbers, CallbackInfo ci) {
+        if ((Object) this instanceof VillagerEntity) {
+            VillagerEntity villager = (VillagerEntity) (Object) this;
+            SpongeVillagerRegistry.getInstance().populateOffers((Villager) this, (List<TradeOffer>) (List<?>) impl$newOffers, (Profession) villager.getVillagerData().getProfession(), villager.getVillagerData().getLevel(), this.rand);
+            givenMerchantOffers.addAll(this.impl$newOffers); // Finally add the mutated offers to the trade offer map.
+            this.impl$newOffers.clear(); // And clean up our temp values
+        }
+    }
+
+    /**
+     * @author i509VCB - February 17th, 2020 - 1.14.4
+     *
+     * @reason In order to apply the Trade mutators, we need to redirect addition to the MerchantOffers till the all offers have been "created".
+     * @param merchantOffers The current offers the villager has.
+     * @param offer The merchant offer to add.
+     * @return true.
+     */
+    @Redirect(method = "addTrades(Lnet/minecraft/item/MerchantOffers;[Lnet/minecraft/entity/merchant/villager/VillagerTrades$ITrade;I)V",
+        at = @At(value = "INVOKE", target = "Lnet/minecraft/item/MerchantOffers;add(Lnet/minecraft/item/MerchantOffer;)Z")
+    )
+    private boolean impl$addNewOfferToTempMap(MerchantOffers merchantOffers, MerchantOffer offer) {
+        if ((Object) this instanceof VillagerEntity) { // Currently only villagers support trade mutators.
+            return this.impl$newOffers.add(offer);
```

Comment:
Same with this, if the logic is specific to `VillagerEntity`, it can be overridden in the mixin